Done

rmullig2 opened the pull request.

cernanb commented on app/controllers/application_controller.rb:
```diff
@@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
+    enable :sessions
+    set :session_secret, "fwitter"
+  end
+  
```
This controller is overloaded. Consider separating your concerns by splitting some of these actions into separate controllers (i.e. a users controller and a tweets controller).

cernanb commented on app/controllers/application_controller.rb:
```diff
@@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
+  get '/' do
+    erb :index
+  end
+  
```
Consider creating helper methods in your ApplicationController to grab the current user and to determine if a user is logged in. 

```
helpers do
    def logged_in?
      !!session[:user_id]
    end

    def current_user
      User.find(session[:user_id])
    end

  end
```

cernanb commented on app/controllers/application_controller.rb:
```diff
@@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
+  end
+  
+  get '/signup/?' do
+    if session[:id].nil?
```
You can use your `logged_in?` helper method here, which is preferable to referring to the session object directly.

cernanb commented on app/controllers/application_controller.rb:
```diff
@@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
+    end
+  end
+  
+  post '/tweets/new' do
```
Typically, tweet creation would be posted to `/tweets` instead of `/tweets/new`.

cernanb commented on app/controllers/application_controller.rb:
```diff
@@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
+  end
+  
+  post '/tweets/new' do
+    if !params[:content].empty?
```
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html#presence) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.

cernanb commented on app/controllers/application_controller.rb:
```diff
@@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
+  post '/tweets/new' do
+    if !params[:content].empty?
+      @user = User.find_by params[:user_id]
+      Tweet.create(params)
```
Another way to do this is to make use of the `has_many` relationship on User:
```
@tweet = current_user.tweets.create(content: params[:content])
```
Both approaches are good, but in some cases you may find this option useful.

cernanb commented on app/controllers/application_controller.rb:
```diff
@@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
+    #binding.pry
+    @user = User.find_by id: session[:id]
+    if !@user.nil?
+      @tweet = Tweet.find(params[:tweet_id])
```
Shouldn't you verify a tweet belongs to the `current_user` (and not just that any user is logged in) before allowing them to edit?

cernanb commented on app/controllers/application_controller.rb:
```diff
@@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
+    #binding.pry
+  end
+  
+  post '/tweets/change' do
```
Consider using the `patch` HTTP verb in this update action. Also, convention for this route would be `'/tweets/:tweet_id'`.

rmullig2 commented:
Cernan,

I have modified my project to include your suggestions. When you have a 
chance, take a look and see if there is anything else that needs to be 
changed.


Thanks,


Ray


On Thu, Dec 08, 2016 at 01:46 PM, cernanb wrote:

@cernanb commented on this pull request.


Hi Ray,

Great job on your Fwitter project. I did a code review on Github, and 
left some comments for you. You should have received some notifications 
from Github. Please notify me when you have completed the fixes or if 
you have any questions by email: cernan@flatironschool.com 
<mailto:cernan@flatironschool.com>  or on slack: (@cernanb 
<https://github.com/cernanb> ). If you would like to schedule a 1 on 1 
session to go over these updates let me know.

Regards,

Cernan Bernardo


In app/controllers/application_controller.rb 
<https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/247#pullrequestreview-12084445> 
:
> @@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
    configure do
      set :public_folder, 'public'
      set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "fwitter"
+  end
+


This controller is overloaded. Consider separating your concerns by 
splitting some of these actions into separate controllers (i.e. a users 
controller and a tweets controller).




In app/controllers/application_controller.rb 
<https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/247#pullrequestreview-12084445> 
:
> @@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
    configure do
      set :public_folder, 'public'
      set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "fwitter"
+  end
+
+  get '/' do
+    erb :index
+  end
+


Consider creating helper methods in your ApplicationController to grab 
the current user and to determine if a user is logged in.
helpers do
     def logged_in?
       !!session[:user_id]
     end

     def current_user
       User.find(session[:user_id])
     end

   end





In app/controllers/application_controller.rb 
<https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/247#pullrequestreview-12084445> 
:
> @@ -5,6 +5,157 @@ class ApplicationController < Sinatra::Base
    configure do
      set :public_folder, 'public'
      set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "fwitter"
+  end
+
+  get '/' do
+    erb :index
+  end
+
+  get '/signup/?' do
+    if session[:id].nil?


You can use your logged_in? helper method here, which is preferable to 
referring to the session object directly.




In app/controllers/application_controller.rb 
<https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/247#pullrequestreview-12084445> 
:
> +    end
+  end
+
+  post '/login' do
+    @user = User.find_by username: params[:username]
+    #binding.pry
+    if !@user.nil? && @user.authenticate(params[:password]) != false
+      session[:id] = @user.id
+      #binding.pry
+      redirect to "/tweets"
+    else
+      redirect to "/login"
+    end
+  end
+
+  post '/tweets/new' do


Typically, tweet creation would be posted to /tweets instead of 
/tweets/new.




In app/controllers/application_controller.rb 
<https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/247#pullrequestreview-12084445> 
:
> +  end
+
+  post '/login' do
+    @user = User.find_by username: params[:username]
+    #binding.pry
+    if !@user.nil? && @user.authenticate(params[:password]) != false
+      session[:id] = @user.id
+      #binding.pry
+      redirect to "/tweets"
+    else
+      redirect to "/login"
+    end
+  end
+
+  post '/tweets/new' do
+    if !params[:content].empty?


I think you could use ActiveRecord validations 
<http://guides.rubyonrails.org/active_record_validations.html#presence> 
(e.g. validates_presence_of :username) to perform these types of checks 
more easily.




In app/controllers/application_controller.rb 
<https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/247#pullrequestreview-12084445> 
:
> +  post '/login' do
+    @user = User.find_by username: params[:username]
+    #binding.pry
+    if !@user.nil? && @user.authenticate(params[:password]) != false
+      session[:id] = @user.id
+      #binding.pry
+      redirect to "/tweets"
+    else
+      redirect to "/login"
+    end
+  end
+
+  post '/tweets/new' do
+    if !params[:content].empty?
+      @user = User.find_by params[:user_id]
+      Tweet.create(params)


Another way to do this is to make use of the has_many relationship on 
User:
@tweet = current_user.tweets.create(content: params[:content])


Both approaches are good, but in some cases you may find this option 
useful.




In app/controllers/application_controller.rb 
<https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/247#pullrequestreview-12084445> 
:
> +
+  get '/tweets/:tweet_id' do
+    if !session[:id].nil?
+      @tweet = Tweet.find_by params[:tweet_id]
+      erb :'/tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+    #binding.pry
+  end
+
+  get '/tweets/:tweet_id/edit' do
+    #binding.pry
+    @user = User.find_by id: session[:id]
+    if !@user.nil?
+      @tweet = Tweet.find(params[:tweet_id])


Shouldn't you verify a tweet belongs to the current_user (and not just 
that any user is logged in) before allowing them to edit?




In app/controllers/application_controller.rb 
<https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/247#pullrequestreview-12084445> 
:
> +    end
+  end
+
+  post '/tweets/new' do
+    if !params[:content].empty?
+      @user = User.find_by params[:user_id]
+      Tweet.create(params)
+      #binding.pry
+      erb :'/tweets/show_user_tweets'
+    else
+      redirect to '/tweets/new'
+    end
+    #binding.pry
+  end
+
+  post '/tweets/change' do


Consider using the patch HTTP verb in this update action. Also, 
convention for this route would be '/tweets/:tweet_id'.


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub 
<https://github.com/learn-co-students/sinatra-fwitter-group-project-v-000/pull/247#pullrequestreview-12084445> 
, or mute the thread 
<https://github.com/notifications/unsubscribe-auth/AHeHQFShUB-fBqI6XtQANxEGUaWFFuezks5rGFCAgaJpZM4LHX-y> 
.
